### PR TITLE
add simplesok

### PR DESCRIPTION
**Pull request by jasonmacduffie**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Note: I am in contact with the upstream developer for Simple Sokoban who is fine with this.

### CODE REVIEW

File: io.osdn.simplesok.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - make
+      - mkdir -p /app/bin
+      - mkdir -p /app/share/simplesok
+      - mv simplesok /app/bin/simplesok
+      - mv skins /app/share/simplesok/skins
+    sources:
+      - type: archive
+        url: http://simplesok.osdn.io/simplesok-1.0.3.tar.xz
+        sha256: cbfaa6b1befc907652b9420220376377a43e9668131dd18573c7caa9678bc610
+finish-args:
```
Comment (hfiguiere) [inline io.osdn.simplesok.yaml:20]: finish-args is usuall found above `modules`.

**Comment by hfiguiere** [comment]
this is mssing the required .desktop, appstream and icon files. I don't see them installed by your build steps.

File: io.osdn.simplesok.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+command: simplesok
+modules:
+  - name: simplesok
+    buildsystem: simple
+    build-commands:
+      - sed -i 's/usr\/share\/simplesok/app\/share\/simplesok/g' skin.c
+      - make
+      - mkdir -p /app/bin
+      - mkdir -p /app/share/simplesok
+      - mv simplesok /app/bin/simplesok
```
Comment (hfiguiere) [inline io.osdn.simplesok.yaml:14]:
```suggestion
      - install -Dm755 simplesok /app/bin/simplesok
      - mkdir -p /app/share/simplesok
```

**Comment by jasonmacduffie** [comment]
> this is mssing the required .desktop, appstream and icon files. I don't see them installed by your build steps.

I missed these requirements, I am working on adding these now.

**Comment by jasonmacduffie** [comment]
All issues should be resolved now, and I will ask the upstream dev to include the new files.

File: io.osdn.simplesok.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      - mkdir -p /app/share/simplesok
+      - mv skins /app/share/simplesok/skins
+      - install -D io.osdn.simplesok.metainfo.xml /app/share/metainfo/io.osdn.simplesok.metainfo.xml
+      - install -D io.osdn.simplesok.desktop /app/share/applications/io.osdn.simplesok.desktop
+      - install -D io.osdn.simplesok.svg /app/share/icons/hicolor/scalable/apps/io.osdn.simplesok.svg
+    sources:
+      - type: archive
+        url: http://simplesok.osdn.io/simplesok-1.0.3.tar.xz
+        sha256: cbfaa6b1befc907652b9420220376377a43e9668131dd18573c7caa9678bc610
+      - type: file
```
Comment (hfiguiere) [inline io.osdn.simplesok.yaml:29]: have these files been submitted upstream? It's a bug then don't ship any.

File: io.osdn.simplesok.desktop [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Name=Simple Sokoban
+GenericName=Simple Sokoban
+Comment=Simple Sokoban Game
+Icon=simplesok
```
Comment (hfiguiere) [inline io.osdn.simplesok.desktop:6]: this is the wrong icon name.

File: io.osdn.simplesok.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>MIT-Modern-Variant</project_license>
+  
+  <description>
+    <p>
+      Simple Sokoban is a Sokoban game aimed at playability and portability across systems.
+    </p>
+  </description>
+  
+  <launchable type="desktop-id">simplesok.desktop</launchable>
```
Comment (hfiguiere) [inline io.osdn.simplesok.metainfo.xml:17]: this is the wrong launchable.

File: io.osdn.simplesok.yaml (hunk 3 above)
Comment (hfiguiere) [reply io.osdn.simplesok.yaml:29]: even since the tarball ship what is needed for rpm and deb. It's all generic for Linux desktop.

Comment (jasonmacduffie) [reply io.osdn.simplesok.yaml:29]: I didn't understand your first comment -- are you saying I should wait until this is merged upstream and a new tarball is created, before continuing?

Comment (hfiguiere) [reply io.osdn.simplesok.yaml:29]: no, but having it in the next release would be nice.

File: io.osdn.simplesok.desktop [hunk 6]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Name=Simple Sokoban
+GenericName=Simple Sokoban
+Comment=Simple Sokoban Game
+Icon=io.osdn.simplesok.svg
```
Comment (hfiguiere) [inline io.osdn.simplesok.desktop:6]:
```suggestion
Icon=io.osdn.simplesok
```

icon name, not file name.

**Comment by hfiguiere** [comment]
bot, build io.osdn.simplesok

**Comment by barthalion** [comment]
Is there anything missing here? It's still marked as draft.

**Comment by jasonmacduffie** [comment]
We are good now.

**Pull request closed without merging**